app/src/main/java/com/kickstarter/viewmodels/CreatorDashboardReferrerBreakdownHolderViewModel.java

```diff
@@ -0,0 +1,270 @@
+import java.util.List;
+
+import rx.Observable;
+import rx.subjects.PublishSubject;
+
+public interface CreatorDashboardReferrerBreakdownHolderViewModel {
+
+  interface Inputs {
+    void projectAndStatsInput(Pair<Project, ProjectStatsEnvelope> projectAndStats);
+  }
+
+  interface Outputs {
+    Observable<Integer> customReferrerColor();
+    Observable<Float> customReferrerPercent();
+    Observable<String> customReferrerPercentText();
+    Observable<Float> customReferrerPledgedAmount();
+    Observable<Integer> externalReferrerColor();
+    Observable<Float> externalReferrerPercent();
+    Observable<String> externalReferrerPercentText();
+    Observable<Integer> internalReferrerColor();
+    Observable<Float> internalReferrerPercent();
+    Observable<String> internalReferrerPercentText();
+    Observable<Boolean> pledgedViaCustomLayoutIsGone();
+    Observable<Boolean> pledgedViaExternalLayoutIsGone();
+    Observable<Boolean> pledgedViaInternalLayoutIsGone();
+    Observable<Pair<Project, Integer>> projectAndAveragePledge();
+    Observable<Pair<Project, Float>> projectAndCustomReferrerPledgedAmount();
+    Observable<Pair<Project, Float>> projectAndExternalReferrerPledgedAmount();
+    Observable<Pair<Project, Float>> projectAndInternalReferrerPledgedAmount();
+  }
+
+  final class ViewModel extends ActivityViewModel<CreatorDashboardReferrerBreakDownViewHolder> implements Inputs, Outputs {
+
+    public ViewModel(final @NonNull Environment environment) {
+      super(environment);
+
+      final Observable<Project> currentProject = this.projectAndProjectStatsInput
+        .map(PairUtils::first);
+
+      final Observable<List<ProjectStatsEnvelope.ReferrerStats>> referrerStats = this.projectAndProjectStatsInput
+        .map(PairUtils::second)
+        .map(ProjectStatsEnvelope::referralDistribution);
+
+      final Observable<ProjectStatsEnvelope.CumulativeStats> cumulativeStats = this.projectAndProjectStatsInput
+        .map(PairUtils::second)
+        .map(ProjectStatsEnvelope::cumulative);
+
+      final Observable<List<ProjectStatsEnvelope.ReferrerStats>> internalReferrers = referrerStats
+        .flatMap(rs ->
+          Observable.from(rs).filter(r -> r.referrerTypeEnumType() == ReferrerType.INTERNAL).toList()
+        );
+
+      final Observable<List<ProjectStatsEnvelope.ReferrerStats>> externalReferrers = referrerStats
+        .flatMap(rs ->
+          Observable.from(rs).filter(r -> r.referrerTypeEnumType() == ReferrerType.EXTERNAL).toList()
+        );
+
+      final Observable<List<ProjectStatsEnvelope.ReferrerStats>> customReferrers = referrerStats
+        .flatMap(rs ->
+          Observable.from(rs).filter(r -> r.referrerTypeEnumType() == ReferrerType.CUSTOM).toList()
+        );
+
+      this.projectAndAveragePledge = cumulativeStats
+        .map(ProjectStatsEnvelope.CumulativeStats::averagePledge)
+        .map(Float::intValue)
+        .compose(avg -> Observable.combineLatest(currentProject, avg, Pair::create));
```

Comment:
rather than doing this `.combineLatest` within a `compose`, it'd be more readable to make this `avg` value an intermediate `Observable<Float>` variable, then the whole output would read as:

```java
this.projectAndAveragePledge = Observable.combineLatest(currentProject, averagePledge, Pair::create);
```